lib/jmap/mdn/disposition.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'disposition.g.dart';
+
+@JsonSerializable()
+class Disposition with EquatableMixin {
+  final ActionMode actionMode;
+  final SendingMode sendingMode;
+  final DispositionType type;
+
+  Disposition({
+    required this.actionMode,
+    required this.sendingMode,
+    required this.type,
+  });
+
+  @override
+  List<Object?> get props => [actionMode, sendingMode, type];
+
+  factory Disposition.fromJson(Map<String, dynamic> json) =>
+      _$DispositionFromJson(json);
+
+  Map<String, dynamic> toJson() => _$DispositionToJson(this);
+}
+
+enum ActionMode {
+  manual('manual-action'),
+  automatic('automatic-action');
+
+  final String value;
+
+  const ActionMode(this.value);
```

Comment:
As the documentation describes, this variable must refer to the values described in the document. So you don't need to create a constructor for it.

<img width="671" alt="Screen Shot 2022-10-07 at 10 42 00" src="https://user-images.githubusercontent.com/80730648/194463785-1bc3e561-768e-4c29-942b-c8f80ed3310c.png">